tests/lib/v9-rule-migration/v9-rule-migration.js

```diff
@@ -0,0 +1,286 @@
+/**
+ * @fileoverview Tests for v9-rule-migration transform.
+ * @author Nitin Kumar
+ * MIT License
+ */
+
+"use strict";
+
+const path = require("path");
+const { applyTransform } = require("@hypermod/utils");
+const assert = require("assert");
+const sinon = require("sinon");
+const { normalizeLineEndngs } = require("../../utils");
+const v9MigrationTransform = require("../../../lib/v9-rule-migration/v9-rule-migration");
+
+describe("v9 migration transform", () => {
+    it("should migrate deprecated context methods to new properties", async () => {
+        const result = await applyTransform(
+            v9MigrationTransform,
+            `
+            module.exports = {
+                create(context) {
+                    return {
+                        Program(node) {
+                            const sourceCode = context.getSourceCode();
+                            const cwd = context.getCwd();
+                            const filename = context.getFilename();
+                            const physicalFilename = context.getPhysicalFilename();
+                        },
+                    };
+                }
+            };
+            `
+        );
+
+        assert.strictEqual(
+            normalizeLineEndngs(result),
+            normalizeLineEndngs(
+                `
+            module.exports = {
+                create(context) {
+                    const sourceCode = context.sourceCode ?? context.getSourceCode();
+                    return {
+                        Program(node) {
+                            const sourceCode = sourceCode;
```

Comment:
It seems like if the name on the left and on the right end up the same, we can just omit the declaration altogether?

Reply:
Yes, updated 👍🏻